version: '3.4'

volumes:
  pimcore-db:
    driver: local
  pimcore-php:
    driver: local

services:
  pimcore:
    build:
      context: ./build
      target: dev
    image: trydirect/pimcore:6-dev
    container_name: pimcore
    hostname: pimcore
    restart: always
    environment:
      - "MYSQL_HOST=db"
      - "SYMFONY_ENV=${SYMFONY_ENV:-dev}"
      - "XDEBUG_PROFILER_ENABLED=${XDEBUG_PROFILER_ENABLED:-0}"
      - "XDEBUG_REMOTE_AUTOSTART=${XDEBUG_REMOTE_AUTOSTART:-1}"
      - "XDEBUG_REMOTE_ENABLE=${XDEBUG_REMOTE_ENABLE:-1}"
      - "XDEBUG_REMOTE_HOST=${XDEBUG_REMOTE_HOST:-host.docker.internal}"
      - "XDEBUG_REMOTE_PORT=${XDEBUG_REMOTE_PORT:-9001}"
      - "XDEBUG_IDEKEY=${XDEBUG_IDEKEY:-PHPSTORM}"
      - "IDE_CONFIG=${IDE_CONFIG:-serverName=pimcore}"
    volumes:
      - "pimcore-php:/var/www/html/"
    links:
      - db

  db:
    image: mysql:5.7
    container_name: pimcoredb
    restart: always
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      timeout: 30s
      retries: 10
    volumes:
      - ./database/initial-dump.sql:/docker-entrypoint-initdb.d/initial_dump.sql
      - ./build/scripts:/scripts
      - pimcore-db:/var/lib/mysql
    env_file:
      - .env

  nginx:
      image: nginx:1.16
      container_name: nginx
      volumes:
        - ./build/configs/nginx/nginx.conf:/etc/nginx/nginx.conf
        - ./build/configs/nginx/conf.d:/etc/nginx/conf.d
        - "pimcore-php:/var/www/html"
      depends_on:
        - pimcore
      ports:
        - "80:8080"
      user: "2000"
